File: 2024/09 - September/19. Reverse Words/Solution.py
class Solution:
    def reverseWords(self,str):
        words = str.split('.')
        words.reverse()
        
        res = ''
        for i in range(len(words)):
            res += words[i]
            if i != len(words) - 1:
                res += '.'
        
        return res

File: 2024/09 - September/19. Reverse Words/Solution.cpp
class Solution 
{
  public:
    string reverseWords(string str) 
    {
        stringstream ss(str);
        string word;
        vector<string> words;
        
        while (getline(ss, word, '.'))
        {
            words.push_back(word);
        }
        
        reverse(words.begin(), words.end());
        
        string res;
        for(size_t i=0; i < words.size(); i++)
        {
            res += words[i];
            if (i != words.size() - 1) 
            {
                res += ".";
            }
        }
        
        return res;
    }
};
